packages/react/select/src/Select.tsx

```diff
@@ -426,36 +438,69 @@ type SelectContentContextValue = {
     value: string,
     disabled: boolean
   ) => void;
+  focusSelectedItem?: () => void;
   selectedItemText?: SelectItemTextElement | null;
-  onScrollButtonChange?: (node: SelectScrollButtonImplElement | null) => void;
   isPositioned?: boolean;
-  shouldExpandOnScrollRef?: React.RefObject<boolean>;
   searchRef?: React.RefObject<string>;
 };
 
 const [SelectContentProvider, useSelectContentContext] =
   createSelectContext<SelectContentContextValue>(CONTENT_NAME);
 
-type SelectContentImplElement = React.ElementRef<typeof DismissableLayer>;
-type DismissableLayerProps = Radix.ComponentPropsWithoutRef<typeof DismissableLayer>;
+const CONTENT_IMPL_NAME = 'SelectContentImpl';
+
+type SelectContentImplElement = SelectPopperOutsideElement | SelectPopperAboveElement;
+type DismissableLayerProps = React.ComponentPropsWithoutRef<typeof DismissableLayer>;
 type FocusScopeProps = Radix.ComponentPropsWithoutRef<typeof FocusScope>;
+
+type SelectPopperPrivateProps = { onPlaced?: PopperContentProps['onPlaced'] };
+
 interface SelectContentImplProps
-  extends Omit<
-    DismissableLayerProps,
-    'disableOutsidePointerEvents' | 'onFocusOutside' | 'onInteractOutside' | 'onDismiss'
-  > {
+  extends Omit<SelectPopperOutsideProps, keyof SelectPopperPrivateProps>,
+    Omit<SelectPopperAboveProps, keyof SelectPopperPrivateProps> {
   /**
    * Event handler called when auto-focusing on close.
    * Can be prevented.
    */
   onCloseAutoFocus?: FocusScopeProps['onUnmountAutoFocus'];
+  /**
+   * Event handler called when the escape key is down.
+   * Can be prevented.
+   */
+  onEscapeKeyDown?: DismissableLayerProps['onEscapeKeyDown'];
+  /**
+   * Event handler called when the a `pointerdown` event happens outside of the `DismissableLayer`.
+   * Can be prevented.
+   */
+  onPointerDownOutside?: DismissableLayerProps['onPointerDownOutside'];
+
+  position?: 'outside' | 'above';
```

Comment:
I'm not sold on `integrated`. I thought about this quite a lot, but it's quite hard to find a balance between:
- something that describes it opens outside like a popper, but can't be called "dropdown", "bottom", "below", etc as we don't know which direction it will open (hence why I went with "outside")
- something that describes it opens "on top" (felt too similar to "top") so went with "above"

Maybe "layered"? What were your thoughts for "integrated"?

Reply:
Maybe `"over"`?

Reply:
Haha cheers for chiming in @atomiks!
It's tough! "over" also could be confused with "above"/"top" although it does carry a bit more meaning.
What do you think @andy-hook?

Reply:
I was thinking `integrated` because the content is "integrated" with the trigger and has strong opinions on the position and layout.

I do like `over` the most though, it adds a bit more meaning (to my mental model at least)

I'd be happy to move forward with that.

Reply:
`zIndex={0}`/`zIndex={1}` 🤣 

On a serious note, I think both `above` and `over` will have the exact same _perceived meaning_. But I think `over` works quite well. Another option: `stacked`

Alternatively, we can consider a boolean prop: `renderOutside` — or something along these lines.

Reply:
Interesting, considering the default behaviour will still be the MacOS way, having a single boolean prop to opt out of it might make some sense?

Maybe something like `asPopper`?

Reply:
`positionOutside`?

Reply:
Is `dropdown` too vague? `<Select dropdown />`

Reply:
Maybe `asDropdown`? That might not be a bad shout as the way most people would ask for this specific feature was: "How do I make the Select position like the DropdownMenu one?"

Reply:
My problem with `asXX` is that we've established it as a convention for modifying the underlying node (`asChild`), we also have a `DropdownMenu` primitive which could be confusing? I think our naming should have relation to other positioning type props e.g. `hideWhenDetached`, `avoidCollisions`.

Reply:
Yep, I agree with the `asXX` sentiment there. We could consider `popper`. It's inline with `modal` on Dialog. Component vs Behaviour kinda thing.

Oooooor... `position="dropdown" | "overlay"`? 😬